### ISSUE TITLE
Mute app sound

### ISSUE THREAD

**Issue by kuznetsov-from-wonderland** [dteoh/SlowQuitApps#45]
Dear developer! I like this app - this helps me in everyday life. Are there any options to mute the sound? Be default, when I close an app - it plays for me a **Bosso** sound from default user-interface-sounds. But this is kind annoying. I can mute all system sounds with F10 on keyboard - but this is not the way for me)

**Comment by kromsik** [comment]
<img width="780" alt="screenshot 2018-11-09 at 08 17 32" src="https://user-images.githubusercontent.com/33729094/48244756-4becba00-e3f8-11e8-8138-1b3365639235.png">

**Comment by dteoh** [comment]
That is very strange. There is no code in SQA that intentionally plays a sound. I have not heard that sound effect go off before...

**Comment by sidebog7** [comment]
It only seems to do it on certain applications. Twitterrific, ImageOptim and OpenEmu seems to do it for me. If you hold Command+Q down it starts by making the sound and then, with a second pause, it starts the constant repeated beginning of the sound. (This is basically the sound that occurs when input is not accepted, an example of this being if you hold a key down in Disk Utility).
I have a video with the sound happening but don't seem to be able to upload it.

**Comment by dnblankedelman** [comment]
So, I have also seen this behavior. My best guess is that instead of SQA successfully trapping the keypresses in some situations and not passing them on to the underlying app, the app receives a bunch of "Qs" it doesn't know how to handle and hence repeatedly beeps (thanks to the keyboard autorepeat).

**Comment by JayBrown** [comment]
We've already had this [here](https://github.com/dteoh/SlowQuitApps/issues/42), and it has continued on to Mojave (using v0.6.0). I'm getting the constant alert sound e.g. in Nimble Commander and BBEdit, but not in Firefox, iTerm, Mail.

The only solution currently is to revert back to v0.5.0, even on Mojave.

**Comment by JayBrown** [comment]
Addendum: Apple Mail now also with the alert sound, and Firefox has one short alert sound at the end right before actually quitting. So this needs to be addressed.

**Comment by dteoh** [comment]
Okay, I finally am able to reproduce this on demand. I tend to press and hold "Cmd" first, then "Q", so I don't hear anything.

If I do the reverse "Q" then "Cmd", I can make the beep sound appear.

Sounds similar to #43.

**Comment by JayBrown** [comment]
I have to add that this is how I press CMD-Q too: first press & hold CMD, then shortly afterwards Q, and then hold both. And the alert sounds also occur when doing it in this order. (Not with v0.5.0, though.)

**Comment by dteoh** [comment]
Please give v0.7.0 a go, it *might* fix this issue...

**Comment by JayBrown** [comment]
Sorry to inform you that the constant alert sound while quitting still happens, e.g. when quitting Firefox, iTerm etc.

**Comment by dteoh** [comment]
I cannot reproduce this anymore, I suspect that there might be some interference with other global key bindings or some issue particular to the local setup.

**Comment by JayBrown** [comment]
This only seems to happen when the application in question (Firefox, iTerm etc.) has no open Windows. I.e. if you have a Firefox window or iTerm window open, and you press CMD-Q, the alert sound doesn't occur. If you close the windows, and the app is still active, and when you then press CMD-Q, then the alert sounds occur. (It doesn't seem to occur when all windows are minimized.)

Maybe that can help you in finding the bug.

**Comment by JayBrown** [comment]
But overall, the behavior is rather erratic… sometimes the system (the apps) behave as described above, then at some point it's different again, sometimes only for a short time.

**Comment by JayBrown** [comment]
I'm always getting an "invalid event source" error from SlowQuitApps, sometimes accompanied by alert sounds, sometimes not.

![snap](https://i.imgur.com/DP8MWsk.jpg)

**Comment by anugupta** [comment]
Some additional info – this happens to me when I'm running BetterTouchTools. 

If I quit BTT, the I no longer get the Basso beep when cmd-q from apps.

**Comment by JayBrown** [comment]
This could be the reason. You first have to quit both BTT and then SlowQuitApps, the latter with `osascript -e 'tell application "SlowQuitApps" to quit'`, then relaunch SlowQuitApps. Then the macOS alert sound will be gone. But I need to test this for a couple of hours without BTT.

When I didn't quit & relaunch SlowQuitApps, then I still got the alert sound.

**Comment by JayBrown** [comment]
Nope. Just opened another app, and got the alert sound too, even with BTT disabled. Then I waited a couple of seconds, and then the alert sound was gone (same app). So I'm not sure if it's BTT after all. Seems to be very erratic/chaotic behavior on SQA's part.

**Comment by JayBrown** [comment]
I may have narrowed it down a little bit. Version 0.7.2 is working fine for at least a minute or so, but then the alert sounds return, sometimes faster when I open another app after launching SQA, e.g. the aforementioned BBEdit. When I then quit SQA with osascript and relaunch it, then everything is fine again, but only for a short period. v0.5.0 keeps working fine, though it might break on Catalina… not sure. (Tested on Mojave 10.14.5.)

**Comment by cameronsstone** [comment]
FYI: I'm on 0.7.3 and I get the sound occasionally. I just got it in firefox immediately after coming out of sleep, but then it stopped after a few seconds/represses...

**Comment by JayBrown** [comment]
In my case, Finder isn't even running. :scratch-head:

**Comment by nassi** [comment]
The beep is intermittent.  I've tried it with/without Finder running, with/without other apps.

**Comment by lucianodinapoli** [comment]
Perhaps this could be solved with the other feature suggestion: press cmd+Q once, the timer comes up and it can be cancelled with ESC. This way we won't be holding the keys and there won't be any obnoxious sounds or errors.

**Comment by nassi** [comment]
This could work.  But isn’t there a way to just swallow the cmd+Q and not pass it on?

---
Ike Nassi

email:       nassi@nassi.com
mobile:     +1-408-390-8281
web:         http://www.nassi.com

**Comment by jvlobo** [comment]
This is happening to me also now since I updated it from version 0.2.1 as specified here: https://github.com/dteoh/SlowQuitApps/issues/77

It is not happening every time so I'm not too sure what's the actual way of reproducing the issue.
I guess it was not happening before because I didn't need to keep `Q` pressed.

**Comment by lucianodinapoli** [comment]
> This is happening to me also now since I updated it from version 0.2.1 as specified here: #77
> 
> It is not happening every time so I'm not too sure what's the actual way of reproducing the issue.
> I guess it was not happening before because I didn't need to keep `Q` pressed.
> 
> Here you can see how I reproduce it by opening WhatsApp: https://drive.google.com/file/d/1h9qJ9ubnV74gVq9KZUr1REeueMwnDR0J/view?usp=sharing

How do you close apps without holding `cmd` + `Q` then? That is the whole point of the SlowQuitApps...

**Comment by nassi** [comment]
I use Keyboard Maestro.  I redefine ⌘-Q in the keyboard prefs to ^ ⌘ ⇧ Q and I trip ⌘Q using a KM Hot Key trigger


---
Ike Nassi

email:      nassi@nassi.com
mobile:    +1-408-390-8281
web:        http://www.nassi.com

**Comment by JayBrown** [comment]
For those of you still encountering this bug: I have switched to **[CommandQ](https://commandqapp.com/)** by Tyler Hall, and it's working fine on macOS 10.15.6. You need to invest $12, however. (Also comes with a delayed [CMD]-[SHIFT]-W when closing windows.)
